src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2013,13 +2013,13 @@ def list_locations(cmd, sku, linux_workers_enabled=None):
 def _check_zip_deployment_status(deployment_status_url, authorization, timeout=None):
     import requests
     total_trials = (int(timeout) // 2) if timeout else 450
-    for _num_trials in range(total_trials):
+    num_trials = 0
+    while num_trials < total_trials:
         time.sleep(2)
         response = requests.get(deployment_status_url, headers=authorization)
         res_dict = response.json()
         if res_dict.get('status', 0) == 3:
-            logger.warning("Zip deployment failed status %s", res_dict['status_text'])
-            break
+            raise ValueError("Zip deployment failed.")
```

Comment:
This should be a CLIError. You can import it from knack.util. CLIError's are appropriately handled by the CLI, however ValueError's result in stack traces.

Reply:
@adewaleo does this need to be CLIError as well? Thank you.

https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py#L2029

Reply:
Yes, this should be a CLIError. In general, any error you throw (and don't handle) should be a CLIError, because the CLI has custom logic for handling CLIErrors gracefully. And no problem at all.

Reply:
Also is it possible to check the res_dict for any error message and to throw that instead? We could then default to and/ or add deployment failed, deployment taking too long, etc. as appropriate.